src/Solvers/SubSolvers.jl

```diff
@@ -0,0 +1,74 @@
+"""
+    SubSolver
+
+An abstract supertype for structures that contain user-controlled parameters 
+for a linear solver applied to compressed matrix blocks. 
+"""
+abstract type SubSolver end
+
+"""
+    SubSolverRecipe
+
+An abstract supertype for structures that contain user-controlled parameters, linear system
+specific parameters and preallocated memory for a linear solver applied to compressed matrix
+blocks. 
+"""
+abstract type SubSolverRecipe end
+
+# Docstring Components
+sub_solver_arg_list = Dict{Symbol,String}(
+    :sub_solver => "`solver::SubSolver`, a user-specified sub-solving method.",
+    :sub_solver_recipe => "`solver::SubSolverRecipe`, a fully initialized realization for a
+    logging method for a specific linear sub solver.",
+    :A => "`A::AbstractMatrix`, a target matrix for compression.",
+)
```

Comment:
It would be good to undestand why `b` is never an argument in any of the methods described below.